common/display/display.cpp

```diff
@@ -164,8 +164,18 @@ bool Display::GetActiveConfig(uint32_t *config) {
   return true;
 }
 
-bool Display::SetDpmsMode(uint32_t dpms_mode) {
-  return display_queue_->SetDpmsMode(dpms_mode);
+bool Display::SetDpmsMode(uint32_t power_mode) {
+
+  power_mode_ = power_mode;
+  uint32_t dpms_prop = display_queue_->GetDpmsProp();
+
+  if (power_mode == DisplayPowerMode::kOff ||
```

Comment:
Even initially i had it as part of the display queue only.
But the api expectation is such that the transition should happen when the function returns.
From the setPowerMode definition
'Sets the power mode of the given display. The transition must be complete when this function returns.'